### PR TITLE
improve definition of CNI configuration in defaults.yaml

### PR DESCRIPTION
**Pull request by StefanoSalsano**
1) add constant definitions for Vagrantfile error exit codes
2) introduces a safer and more user-friendly approach for configuration options:
in order to define a settings key called "my_conf_key" that can assume the values: "First", "Second", Third", this is the new format that can be used (note the suffix "_option" to be added to the key name:
   my_conf_key_options:
     First: false
     Second: true
     Third: false
the function check_and_select_conf_options(selected_dict,target_key,option_array,error)
checks that one and at most one option has been selected

3) added the possibility to configure Calico tunnel mode from .yaml configuration, using the new mechanism as follows:
    calico_config:
      calico_env_var_options:
         CALICO_IPV4POOL_IPIP: true
         CALICO_IPV4POOL_VXLAN: false
      calico_env_var_value_options:
         Always: true
         CrossSubnet: false
         Never: false

4) modified the .gitignore to cover env*.yaml, it was needed to locally keep multiple variants of the env.yaml customization file

### CODE REVIEW

File: defaults.yaml [hunk 1]
```diff
@@ -76,7 +76,19 @@ ansible:
       # Allowed values are: weavenet, calico, flannel, and no-cni-plugin.
       # If you choose no-cni-plugin, provisioners will prepare the environment
       # but will NOT bootstrap the Kubernetes cluster.
-      kubernetes_network_plugin: calico
+      kubernetes_network_plugin_options:
```
Comment (ferrarimarco) [inline defaults.yaml:79]: Can you please update the comments for this parameters and comments for the ones you added?

File: Vagrantfile [hunk 2]
```diff
@@ -167,6 +205,9 @@ broadcast_address = n | (~n.instance_variable_get(:@mask_addr) & IPAddr::IN4MASK
 cluster_ip_cidr = settings["pod_network"]["cluster_ip_cidr"]
 service_ip_cidr = settings["pod_network"]["service_ip_cidr"]
 
+calico_env_var = settings["ansible"]["group_vars"]["all"]["calico_config"]["calico_env_var"]
+calico_env_var_value = settings["ansible"]["group_vars"]["all"]["calico_config"]["calico_env_var_calue"]
```
Comment (ferrarimarco) [inline Vagrantfile:209]: Typo in `calico_env_var_calue`

File: Vagrantfile [hunk 3]
```diff
@@ -107,15 +133,27 @@ end
...
 log_info_or_debug "Active settings (from defaults.yaml and env.yaml): #{settings.to_yaml}"
 
-# Check that an allowed networking plugin is provided
-allowed_cni_plugins=["weavenet","calico","flannel","no-cni-plugin"]
-if not allowed_cni_plugins.include? settings["ansible"]["group_vars"]["all"]["kubernetes_network_plugin"]
-  @ui.error 'kubernetes_network_plugin is not valid in defaults.yaml or env.yaml, allowed values are:'
-  allowed_cni_plugins.each {|valid| @ui.error valid }
-  exit(1)
+# Check that at least one and only one plugin is selected
+#settings["ansible"]["group_vars"]["all"]["kubernetes_network_plugin"] =
```
Comment (ferrarimarco) [inline Vagrantfile:139]: Remove this commented line

File: Vagrantfile [hunk 4]
```diff
@@ -92,6 +98,26 @@ settings_merger = proc {
     end
 }
 
+def check_and_select_conf_options(selected_dict,target_key,option_array,error)
```
Comment (ferrarimarco) [inline Vagrantfile:101]: Can you add a comment that explains what this function does?

File: Vagrantfile (hunk 2 above)
Comment (ferrarimarco) [reply Vagrantfile:209]: I think we should add a check to prevent this kind of issues. I'll open a new one.

Comment (StefanoSalsano) [reply Vagrantfile:209]: thank you!!

File: defaults.yaml (hunk 1 above)
Comment (StefanoSalsano) [reply defaults.yaml:79]: done!

File: Vagrantfile (hunk 3 above)
Comment (StefanoSalsano) [reply Vagrantfile:139]: done

File: Vagrantfile (hunk 4 above)
Comment (StefanoSalsano) [reply Vagrantfile:101]: done

**Pull request merged**